### ISSUE TITLE
TypeError in balance.py

### ISSUE THREAD

**Issue by RevelBerry** [hearmeneigh/dataset-rising#1]
when running through the README.md steps for this project: [e621-rising-configs](https://github.com/hearmeneigh/e621-rising-configs) there is an TypeError occurring at line 30 of balance.py in the Dataset Rising toolset:

total_ratio = math.fsum([sel.ratio for sel in selections])

It looks like 'fixed' is being passed as the sel.ratio in some cases, where math.fsum() is expecting a 'real number'. 

There seems to be some logic issues in balance.py... specifically:

1. In balance_selections(), the loop 'for sel in selections:' will only ever run 'resolve_fixed_ratios()' once even if more than one 'sel' has a sel.ratio of 'fixed' because of the 'break'. The solution is to remove 'break'
2. In resolved_fixed_ratios(), 'total_dynamic_ratio' can/will be 0, not only when 'everything is *' but also when all of the 'sel.ratio' values are 'fixed'. Solution is to comment out "if total_dynamic_ratio = 0: return" because that line prevents the 'fixed' sel.ratio values from being resolved.

Doing the above resolved the issues I was experiencing trying to run dr-join using only 2 tiers (tier 1 and extras)

**Comment by hearmeneigh** [comment]
@RevelBerry Can you share the command line arguments that you used to produce the error, please?

**Comment by RevelBerry** [comment]
Hi, yes from the e621-rising-configs README I used this:

```
## join selected samples into a single JSONL file
dr-join 
  --samples "${BUILD_PATH}/samples/tier-1.jsonl:*" \
  --samples "${BUILD_PATH}/samples/extras.jsonl:*" \
  --output "${BUILD_PATH}/dataset/samples.jsonl" \
  --export-tags "${BUILD_PATH}/dataset/tag-counts.json" \
  --export-autocomplete "${BUILD_PATH}/dataset/webui-autocomplete.csv" \
  --min-posts-per-tag 100 \
  --min-tags-per-post 15 \
  --prefilter ./dataset/prefilter.yaml
```

The only change for the commands was I removed the tier-2, tier-3 and tier-4 sample jsonl files. I also had made some edits to the includes/excludes in the tier-1 and tier-1 helper files.

**Comment by hearmeneigh** [comment]
Fixed in `1.0.4`
